### DISCUSSION TITLE
Ported wozmon

### DISCUSSION THREAD

**Discussion by quells** [picocomputer/community#48] [Q&A]
Not terribly interesting, but I've ported Ben Eater's port of the WOZ Monitor: https://github.com/quells/wozmon.rp6502

I couldn't figure out how to get the VS Code template to work with assembly, so I wrote a separate assemble-and-upload script using the `vasm` commands. Still uses the `rp6502-sdk` as a submodule for the `ROM` and `Monitor` classes.

**Comment by rumbledethumps** [comment]
What you've made is effectively an Apple I+. An Apple I plus a full featured terminal. Historically interesting and now you can follow along with Ben Eater's video. Good stuff.

**Reply by vruumllc** [reply]
Cool! 
I got this working on Ubuntu linux, but had to:
1: Copy rp6502.py from the rp6502-sdk folder to the upload.py location (couldn't find it otherwise)
2: Change "From rp6502-sdk import rp6502" to just "import rp6502" in upload.py
3: Since I didn't add vasm6502_oldstyle to path, but just copied it to the upload.py location, I had to add "./" in front of vasm6502_oldstyle command in upload.py 

WOZMON requires all caps, so a working CapsLock came just in the nick of time!

**Reply by linuxplayground** [reply]
Check out eWoz that will let you paste in intelhex files to upload.  Also, it automatically converts to uppercase and it has backspace.
Takes up more than a page of ram though. So a bit of a departure from the 249 byte version Steve Wozniak wrote. 
https://gist.github.com/BigEd/2760560

**Comment by snakebyte69** [comment]
I ported Wozmon to Z80 CP/M.
I added a crtl-c to exit.

https://github.com/picocomputer/community/assets/39090848/c8b0bad9-772b-487b-8963-fbe7f487e02b
